### PR TITLE
Added support for Google Custom Search API

### PR DESCRIPTION
**Pull request by russellocean**
### Description
This PR fixes https://github.com/Torantulino/Auto-GPT/issues/31

This PR adds functionality to the project to allow for a choice between the original Google search method and the Google Custom Search API. The google_search method uses the original method of scraping the HTML from the search results page, using googlesearch-python, while the google_official_search method uses the Google Custom Search API to retrieve search results. **Note** the Google Custom Search API provides 100 search queries per day for free. If you need more, you may sign up for [billing](https://cloud.google.com/billing/docs/how-to/manage-billing-account) in the API Console. Additional requests cost $5 per 1000 queries, up to 10k queries per day.

### How to test:

To test the functionality, ensure that you have valid API keys and search engine IDs for both the Google search method and the Google Custom Search API. You can set these values in your environment variables as described in the README.md file.

**Additional Notes:**

This pull request only adds functionality and makes improvements to existing code. No new features or major changes have been introduced.

### CODE REVIEW

**Review by rmonsurate — DISMISSED** [review]
Excellent update

**Comment by robinicus** [comment]
As a heads up, I added these changes in manually to test them but received an error:

`Command google returned: Error: <HttpError 400 when requesting https://customsearch.googleapis.com/customsearch/v1?q=list+of+potential+domain+names+for+ecommerce+health+website&num=8&key=<key>&alt=json  returned "Request contains an invalid argument.". Details: "[{'message': 'Request contains an invalid argument.', 'domain': 'global', 'reason': 'badRequest'}]">`

**Review by rmonsurate — COMMENTED** [review]
Looks good, testing now.

**Comment by rmonsurate** [comment]
Tested without any errors. Did you set the environment variables too?

**Comment by rmonsurate** [comment]
> As a heads up, I added these changes in manually to test them but received an error:
> 
> `Command google returned: Error: <HttpError 400 when requesting https://customsearch.googleapis.com/customsearch/v1?q=list+of+potential+domain+names+for+ecommerce+health+website&num=8&key=<key>&alt=json returned "Request contains an invalid argument.". Details: "[{'message': 'Request contains an invalid argument.', 'domain': 'global', 'reason': 'badRequest'}]">`
> 
> Key is passing in correctly, just removed here obviously. Run with --continuous and --gpt3only for testing purposes.

I'm not seeing any errors on my end. Try adding the environment labels.

**Comment by robinicus** [comment]
Yep, I can confirm this is fixed and working well now as long as I run the setx command in addition to putting the variables in the .env

File: scripts/commands.py [hunk 1]
```diff
@@ -44,7 +47,13 @@ def get_command(response):
 def execute_command(command_name, arguments):
     try:
         if command_name == "google":
-            return google_search(arguments["input"])
+            print("Using Google search method")
```
Comment (Torantulino) [inline scripts/commands.py]:
```suggestion
            
```

**Comment by kvnbln** [comment]
I followed all of the instructions for Google Search and this is error I get.  Please advise.  The variables have been added to the PATH section correctly, but this still isn't working.  Help pelase 

Command google returned: Error: <HttpError 400 when requesting https://customsearch.googleapis.com/customsearch/v1?q=flash+loans+and+current+market+conditions&cx=HollyAveAI&num=8&key=-----------------------&alt=json
 returned "Request contains an invalid argument.". Details: "[{'message': 'Request contains an invalid argument.', 'domain': 'global', 'reason': 'badRequest'}]">

**Comment by kvnbln** [comment]
I followed all of the instructions for Google Search and this is error I get. Please advise. The variables have been added to the PATH section correctly, but this still isn't working. Help pelase

Command google returned: Error: <HttpError 400 when requesting https://customsearch.googleapis.com/customsearch/v1?q=-----------------&cx=HollyAveAI&num=8&key=-----------------------&alt=json
returned "Request contains an invalid argument.". Details: "[{'message': 'Request contains an invalid argument.', 'domain': 'global', 'reason': 'badRequest'}]">

**Comment by GrzegorzKarczewski** [comment]
@kvnbln Check again your search engine id, it should look something like that "c13e46a5286cb3425"

**Comment by TylonHH** [comment]
I set up the Google search API and Engine. But in the search URL is `&key=no` and then I got an invalif key response.

**Comment by andresn** [comment]
Part of this video made the setup simple for me: https://youtu.be/WYPFotzHZ4A?t=180

Perhaps include something like this in the ReadMe for users unfamiliar with Google API services?

**Pull request merged**